### PR TITLE
Update configure script to detect BUILD_ARCH differently, for use by …

### PR DESCRIPTION
**Pull request by johgibso**
…PYCMIX et al.

### CODE REVIEW

File: configure.ac
Comment (dougscott) [inline configure.ac:1]: Does this new configure allow a user to force a x86_64 build on an M1 Mac?  We still want that.

**Comment by johgibso** [comment]
No, because BUILD_ARCH is not passed when building most src files. We never had the capability of forcing a specific architecture. I agree it would be useful.JohnOn Jun 28, 2023, at 4:07 PM, Douglas Scott ***@***.***> wrote:﻿
@dougscott commented on this pull request.



On configure.ac:
Does this new configure allow a user to force a x86_64 build on an M1 Mac?  We still want that.

**Comment by dougscott** [comment]
My question does not deal with BUILD_ARCH -- I am talking about the implementation of these configure options:

System types:
  --build=BUILD     configure for building on BUILD [guessed]
  --host=HOST       cross-compile to build programs to run on HOST [BUILD]

**Comment by dougscott** [comment]
Ah never mind - examining the configure script, I see it does not use "host" ever to set architecture.  Ugh.
OK, I need to make sure I have the tools to build configure so I can try your changes here.

**Comment by dougscott** [comment]
John, I merged the PR and pulled the changes locally and ran configure, and got this error/warning at the end:

configure: error: in `/opt/local/src/RTcmix.git':
configure: error: The pkg-config script could not be found or is too old.  Make sure it
is in your PATH or set the PKG_CONFIG environment variable to the full
path to pkg-config.

**Comment by johgibso** [comment]
Yes, that is a common error when dealing with GNU stuff, if you don't have their pkg-config program installed. But you got this when simply running ./configure, without asking for any options that need pkg-config to discover them?I will temporarily delete my pkg-config to see what happens on my machine. On Jun 28, 2023, at 4:59 PM, Douglas Scott ***@***.***> wrote:﻿
John, I merged the PR and pulled the changes locally and ran configure, and got this error/warning at the end:
configure: error: in `/opt/local/src/RTcmix.git':
configure: error: The pkg-config script could not be found or is too old.  Make sure it
is in your PATH or set the PKG_CONFIG environment variable to the full
path to pkg-config.

**Comment by johgibso** [comment]
I just disabled pkg-config on my system, and made sure with…

~/rtcmix % type pkg-config
pkg-config not found

After this, ./configure continues to run without error.

I wonder what is triggering this error message for you. There should be invocation of pkg-config only if you’re trying to use one of the libraries that require it, namely jack, fftw, and osc.

When I try any of those, I see the error.

The point of pkg-config is to supply compiler flags, library locations, etc. that are cached in .pc files stored in /usr/local/lib/pkgconfig/ and other places. Most any library that is useable on Linux adopts this standard. I don’t want to require it for basic compilation, though, because macOS systems never have pkg-config installed by default.

J

**Comment by johgibso** [comment]
That’s right, and perhaps we should make use of these options!

BUILD_ARCH, which according to the git log of configure.ac, I may be to blame for, doesn’t sound like the right symbol name.

Really what we want is the name of the architecture to build for everything. If that could let us switch between intel and arm for platforms that support both, all the better. Though I’m not sure why one would build x86_64 on Apple Silicon at this point.

**Comment by dougscott** [comment]
Yes I had added --with-osc to my configure options.  I will ignore it.

**Comment by johgibso** [comment]
If you want to build that without having pkg-config, you can pass the cflags and libs to the configure script or put them in makefile.conf, which may be what you did before?

**Pull request merged**